tests/heka/test_message_parser.py

```diff
@@ -85,3 +96,24 @@ def test_invalid_utf8(data_dir):
     with open(filename, "rb") as o:
         for r in message_parser.parse_heka_message(o):
             assert(u'\ufffd' in r['info']['adapterDescription'])
+
+
+def test_lazy_parsing(data_dir, monkeypatch):
+    from moztelemetry.heka.message_parser import _parse_json
+    mock_parse_json = MagicMock(name='_parse_json',
+                                wraps=_parse_json)
+    import moztelemetry.heka.message_parser
+    monkeypatch.setattr(moztelemetry.heka.message_parser, '_parse_json', mock_parse_json)
+
+    filename = "{}/test_telemetry_gzip.heka".format(data_dir)
+    with open(filename, "rb") as o:
+        heka_message = message_parser.parse_heka_message(streaming_gzip_wrapper(o)).next()
+
+    # should only have parsed json *once* (json fields of the message should
```

Comment:
At the very beginning of `_parse_heka_record` in `message_parser.py`. Since the payload forms the skeleton of the dictionary, I suspect there isn't any way to lazily process it.